packages/docusaurus-plugin-content-docs/src/client/docsClientUtils.ts

```diff
@@ -27,14 +27,17 @@ export function getActivePlugin(
   pathname: string,
   options: GetActivePluginOptions = {},
 ): ActivePlugin | undefined {
-  const activeEntry = Object.entries(allPluginDatas).find(
-    ([_id, pluginData]) =>
-      !!matchPath(pathname, {
-        path: pluginData.path,
-        exact: false,
-        strict: false,
-      }),
-  );
+  const activeEntry = Object.entries(allPluginDatas)
+    // A quick route sorting: '/android/foo' should match '/android' instead of '/'
+    .sort((a, b) => b[1].path.localeCompare(a[1].path))
```

Comment:
that works for this case but maybe not all?

We have an existing algo there: packages/docusaurus/src/server/plugins/index.ts#sortConfig

Maybe it could be extracted and reused on a browser/node shared package?

I'd like both algos to stay in sync preferable because it's likely that we'll refactor this later and we'll likely forget to update this file when we do that

Reply:
I essentially took mine from there. There's only one line that really applies here, because we are sorting _paths_, not _routes_

Reply:
ok 👍